pkg/templates/templates.go

```diff
@@ -984,6 +1037,31 @@ func (t *TemplateResolver) GetUsedResources() []unstructured.Unstructured {
 	return t.usedResources
 }
 
+// applyContextTransformers runs the configured ContextTransformers in order and
+// returns the final transformed context. This is only called when a
+// DynamicWatcher is configured.
+func (t *TemplateResolver) applyContextTransformers(
+	origContext interface{},
+	options *ResolveOptions,
+) (interface{}, error) {
+	ctx := origContext
+
+	for i, contextTransformer := range options.ContextTransformers {
+		queryObj := cachingQueryAPI{dynamicWatcher: t.dynamicWatcher, watcher: *options.Watcher}
+
+		newCtx, err := contextTransformer(&queryObj, origContext)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"%w at options.ContextTransformers[%d]: %w", ErrContextTransformerFailed, i, err,
+			)
+		}
+
+		ctx = newCtx
+	}
```

Comment:
I think this is repeatedly overwriting the context and only returning the final transform. I think this was refactored for the SonarCloud issue and I only see one ContextTransformer in the propagator, so it might not matter. Just something I was wondering about.

```suggestion
    var err error
	ctx := origContext

	for i, contextTransformer := range options.ContextTransformers {
		queryObj := cachingQueryAPI{dynamicWatcher: t.dynamicWatcher, watcher: *options.Watcher}

		ctx, err = contextTransformer(&queryObj, ctx)
		if err != nil {
			return nil, fmt.Errorf(
				"%w at options.ContextTransformers[%d]: %w", ErrContextTransformerFailed, i, err,
			)
		}
	}
```